pkgs/applications/networking/cluster/ssm-session-manager-plugin/default.nix

```diff
@@ -1,51 +1,37 @@
-{ stdenv, lib, fetchurl, autoPatchelfHook, dpkg, awscli, unzip }:
-let
-  ver = "1.2.331.0";
-  source = {
-    url = rec {
-      "x86_64-linux" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/ubuntu_64bit/session-manager-plugin.deb";
-      "aarch64-linux" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/ubuntu_arm64/session-manager-plugin.deb";
-      "x86_64-darwin" = "https://s3.amazonaws.com/session-manager-downloads/plugin/${ver}/mac/sessionmanager-bundle.zip";
-      "aarch64-darwin" = x86_64-darwin;
-    }."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
-    sha256 = rec {
-      "x86_64-linux" = "sha256-xWnY89dslkGhRTh9llRFkuUqYIjHQNt+TLnkPQr3u1Q=";
-      "aarch64-linux" = "sha256-QE6+EjLoydTPuLitG6fALXAtvIkfyoFuWij8Z2HT6+Q=";
-      "x86_64-darwin" = "0gr6frdn9jvxnkymkcpvgkqw4z2sac9jdf5qj4hzakq1zkfviazf";
-      "aarch64-darwin" = x86_64-darwin;
-    }."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
-  };
-  archivePath = if stdenv.isDarwin then "sessionmanager-bundle" else "usr/local/sessionmanagerplugin";
-in
-stdenv.mkDerivation rec {
-  pname = "ssm-session-manager-plugin";
-  version = ver;
+{ stdenv
+, lib
+, fetchFromGitHub
+, buildGoPackage
+}:
 
-  src = fetchurl source;
+buildGoPackage rec {
+  pname = "ssm-session-manager-plugin";
+  version = "1.2.398.0";
 
-  nativeBuildInputs = lib.optionals stdenv.isLinux [
-    autoPatchelfHook
-    dpkg
-  ] ++ lib.optionals stdenv.isDarwin [
-    unzip
-  ];
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "session-manager-plugin";
+    rev = version;
+    sha256 = "ufNnr/sxOHUDUsGXwxp1yloVAI6DMtuEdjcQZ2XaHRg=";
+  };
 
-  unpackPhase = if stdenv.isDarwin then "unzip $src" else "dpkg-deb -x $src .";
+  postPatch = ''
+    mv vendor{,-old}
+    mv vendor-old/src vendor
+    rm -r vendor-old
+  '';
 
   installPhase = ''
-    runHook preInstall
-
-    install -m755 -D ${archivePath}/bin/session-manager-plugin $out/bin/session-manager-plugin
-
-    runHook postInstall
+    mkdir -p $out/bin
+    cp "$NIX_BUILD_TOP/go/bin/sessionmanagerplugin-main" $out/bin/session-manager-plugin
```

Comment:
Use of `$NIX_BUILD_TOP` is discouraged: https://github.com/NixOS/nixpkgs/issues/189691